### PR TITLE
Rainbow Option Pricing with bruteforce method using Classiq Platform

### PR DESCRIPTION
**Pull request by giacomoRanieri**
This work completes the novel approach to price rainbow options presented in #103 with a bruteforce method introduced by Or Samimi Golan (@orsa-classiq) 

You can find all the relevant information within the research paper at [this link](https://arxiv.org/pdf/2402.05574v2)

Thanks to the other authors:
- Francesca Cibrario, 
- Or Samimi Golan, 
- Emanuele Dri, 
- Mattia Ippoliti, 
- Ron Cohen, 
- Christian Mattia, 
- Bartolomeo Montrucchio, 
- Amir Naveh, 
- Davide Corbelletto

### CODE REVIEW

File: research/rainbow_options_bruteforce_method/rainbow_options_bruteforce_method.ipynb [hunk 1]
```diff
@@ -0,0 +1,671 @@
+{
```
Comment (orsa-classiq) [inline research/rainbow_options_bruteforce_method/rainbow_options_bruteforce_method.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #3.    #def round_factor(a):
</pre><p>is this one on purpose?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Classiq/classiq-library/pull/175/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (orsa-classiq) [inline research/rainbow_options_bruteforce_method/rainbow_options_bruteforce_method.ipynb:1]: <p>I would explain something about that - that it uses the amplitude loading method. and that it is more of a sanity check for the other algorithms, as general amplitude loading scales exponentially with the number of qubits</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Classiq/classiq-library/pull/175/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

**Review by orsa-classiq — CHANGES_REQUESTED** [review]
Nice:) please see my comments

File: research/rainbow_options_bruteforce_method/rainbow_options_bruteforce_method.ipynb (hunk 1 above)
Comment (giacomoRanieri) [reply research/rainbow_options_bruteforce_method/rainbow_options_bruteforce_method.ipynb:1]: It was definitely something that was left and had to be removed.
I'll push a commit soon with the fix.

Comment (giacomoRanieri) [reply research/rainbow_options_bruteforce_method/rainbow_options_bruteforce_method.ipynb:1]: I've added more information based on your suggestion

**Comment by giacomoRanieri** [comment]
@orsa-classiq Looks like it's failing due to missing credentials during the test phase

**Review by orsa-classiq — APPROVED** [review]

**Pull request merged**